inabox/deploy/env_vars.go

```diff
@@ -543,3 +543,96 @@ func (vars ChurnerVars) getEnvMap() map[string]string {
 	}
 	return envMap
 }
+
+type ControllerVars struct {
+	CONTROLLER_DYNAMODB_TABLE_NAME string
+
+	CONTROLLER_BLS_OPERATOR_STATE_RETRIVER string
+
+	CONTROLLER_EIGENDA_SERVICE_MANAGER string
+
+	CONTROLLER_USE_GRAPH string
+
+	CONTROLLER_ENCODING_PULL_INTERVAL string
+
+	CONTROLLER_AVAILABLE_RELAYS string
+
+	CONTROLLER_ENCODER_ADDRESS string
+
+	CONTROLLER_DISPATCHER_PULL_INTERVAL string
+
+	CONTROLLER_NODE_REQUEST_TIMEOUT string
+
+	CONTROLLER_NUM_CONNECTIONS_TO_NODES string
+
+	CONTROLLER_INDEXER_DATA_DIR string
+
+	CONTROLLER_ENCODING_REQUEST_TIMEOUT string
+
+	CONTROLLER_ENCODING_STORE_TIMEOUT string
+
+	CONTROLLER_NUM_ENCODING_RETRIES string
+
+	CONTROLLER_NUM_RELAY_ASSIGNMENT string
+
+	CONTROLLER_NUM_CONCURRENT_ENCODING_REQUESTS string
+
+	CONTROLLER_FINALIZATION_BLOCK_DELAY string
+
+	CONTROLLER_NUM_REQUEST_RETRIES string
+
+	CONTROLLER_NUM_CONCURRENT_DISPERSAL_REQUESTS string
+
+	CONTROLLER_NODE_CLIENT_CACHE_NUM_ENTRIES string
+
+	CONTROLLER_CHAIN_RPC string
+
+	CONTROLLER_CHAIN_RPC_FALLBACK string
+
+	CONTROLLER_PRIVATE_KEY string
+
+	CONTROLLER_NUM_CONFIRMATIONS string
+
+	CONTROLLER_NUM_RETRIES string
+
+	CONTROLLER_LOG_LEVEL string
+
+	CONTROLLER_LOG_PATH string
+
+	CONTROLLER_LOG_FORMAT string
+
+	CONTROLLER_INDEXER_PULL_INTERVAL string
+
+	CONTROLLER_AWS_REGION string
+
+	CONTROLLER_AWS_ACCESS_KEY_ID string
+
+	CONTROLLER_AWS_SECRET_ACCESS_KEY string
+
+	CONTROLLER_AWS_ENDPOINT_URL string
+
+	CONTROLLER_FRAGMENT_PREFIX_CHARS string
+
+	CONTROLLER_FRAGMENT_PARALLELISM_FACTOR string
+
+	CONTROLLER_FRAGMENT_PARALLELISM_CONSTANT string
+
+	CONTROLLER_FRAGMENT_READ_TIMEOUT string
+
+	CONTROLLER_FRAGMENT_WRITE_TIMEOUT string
+
+	CONTROLLER_GRAPH_URL string
+
+	CONTROLLER_GRAPH_BACKOFF string
+
+	CONTROLLER_GRAPH_MAX_RETRIES string
+}
+
+func (vars ControllerVars) getEnvMap() map[string]string {
+	v := reflect.ValueOf(vars)
+	envMap := make(map[string]string)
+	for i := 0; i < v.NumField(); i++ {
+		envMap[v.Type().Field(i).Name] = v.Field(i).String()
+	}
+	return envMap
+}
```

Comment:
Although I like that this is automated, it's different than how we are reading environment variables elsewhere. Is this a pattern that we use just for inabox, or a new pattern that you think we should adopt elsewhere as well?